core/src/com/unciv/logic/map/InfluenceMap.kt

```diff
@@ -0,0 +1,119 @@
+
+object InfluenceMap {
+
+    fun setInfluenceScore(viewingCiv: CivilizationInfo, tilesToCheck: List<TileInfo>? = null) {
+        val viewable = viewingCiv.gameInfo.tileMap.values
+        
+        //Such a dirty hack, need to actually create an object that holds these for each Civ
+        for (tile in viewable) {
+            tile.influenceScore = 0.0F
+        }
+        val tilesWithEnemies = (tilesToCheck ?: viewable)
+                .filter { containsEnemyUnit(it, viewingCiv) }
+        val tilesWithNeutralCombatant = (tilesToCheck ?: viewable)
+                .filter { containsNeutralUnit(it, viewingCiv) }
+        val tilesWithFriendlyCombatant = (tilesToCheck ?: viewable)
+                .filter { containsFriendlyUnit(it, viewingCiv) }
+        setTiles(tilesWithEnemies)
+        setTiles(tilesWithNeutralCombatant, 0.5F)
+        setTiles(tilesWithFriendlyCombatant, -1.0F)
+
+    }
+    
+    private fun getModifiedUnitStrength(tile: TileInfo, unit: MapUnit): Float {
+        return (unit.baseUnit.strength * (1 + tile.getDefensiveBonus())) * (unit.health/100.0F)
+    }
+    
+    private fun getInfluenceOfUnit(tile: TileInfo, unit: MapUnit, influenceModifier: Float = 1.0F) {
+        if (unit.type.isMilitary()) {
+            tile.influenceScore += getModifiedUnitStrength(tile, unit) * influenceModifier
+            if (unit.type.isMelee()) {
+                val tilesInMoveRange = unit.movement.getDistanceToTilesWithinTurn(
+                        unit.getTile().position,
+                        unit.getMaxMovement().toFloat()
+                ).keys.asSequence().filter { it.position != tile.position }
+                for (tile in tilesInMoveRange) {
+                    tile.influenceScore += getModifiedUnitStrength(tile, unit) * influenceModifier
+                }
+            }
+            if (unit.type.isRanged()) {
+                val rangeOfAttack = unit.getRange()
+                val tilesInTargetRange =
+                        if (unit.hasUnique("Ranged attacks may be performed over obstacles") || unit.type.isAirUnit())
+                            tile.getTilesInDistance(rangeOfAttack)
+                        else tile.getViewableTilesList(rangeOfAttack)
+                                .asSequence()
+                for (tile in tilesInTargetRange) {
+                    tile.influenceScore += getModifiedUnitStrength(tile, unit) * influenceModifier
+                }
+            }
+        } else if (unit.name == Constants.greatGeneral || unit.baseUnit.replaces == Constants.greatGeneral) {
+            val tilesInAffectedRange = tile.getTilesInDistance(2)
+            for (tile in tilesInAffectedRange) {
+                tile.influenceScore *= 1.15F * influenceModifier
+            }
+        } else if (unit.type.isCivilian()) {
+            tile.influenceScore += 4.0F * influenceModifier
+        }
+    }
+    
+    private fun getInfluenceOfCity(city: CityInfo, influenceModifier: Float = 1.0F) {
```

Comment:
This is a function that sets a value, not one that returns a value, and the name should reflect that
setTileCityInfluence, perhaps